src/components/CodeSample/CodeSampleCompile.astro

```diff
@@ -0,0 +1,36 @@
+---
+import CodeSample from "./CodeSample.astro";
+import { tsToJs, getTscOptions } from "./compileTS.ts";
+import { readFile } from "node:fs/promises";
+import CodeSnippet from "../CodeSnippet/CodeSnippet.astro";
+import { Code } from "astro/components";
```

Comment:
Something our code samples do is do some additional formatting like trimming the final line and ensuring tabs, we are not currently exporting it, so I'll need you to do this by changing the integration file (just a matter of adding `export`, no need to change it internally). 

```suggestion
import { Code } from "astro/components";
import { preprocessCode } from "integrations/code-snippets.ts";
```